### DISCUSSION TITLE
RaspberryMatic 3.79.6.20241031

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#2885] [Announcements]
This is release 3.79.6.20241031 of RaspberryMatic which is a **major** release with the following bugfixes and feature changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.79.6.20241031/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.79.6.20241031) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL) [![GitHub sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&link=https://github.com/sponsors/jens-maus)](https://github.com/sponsors/jens-maus)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=83873)</sub>

## 🚧 Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.77.7.20240826...3.79.6.20241031).</sub>

#### CCU/HomeMatic service changes:
- [8f2c8f78f](http://github.com/jens-maus/RaspberryMatic/commit/8f2c8f78f22a0509ead9a78f402737d276e89c56) - updated [OCCU](https://github.com/eq-3/occu) version to [3.79.6-2](https://github.com/jens-maus/occu/tree/b_3_79) integrating full compatibility to the [CCU3 3.79.6 firmware](https://homematic-ip.com/sites/default/files/other_media/ccu3-changelog-3-79-6.pdf) which comes with the following changes:
   - ⚠️ **BREAKING CHANGE** ⚠️: Due to changes in the `HmIPServer` service of the 3.79.x firmware, a downgrade to previous versions may result in lost HmIP devices. Users are therefore highly advised to create a sbk-file Backup previous to any update to 3.79.x in case a downgrade is required to previous versions so that this backup can be restored then.
   - update `ReGaHss` logic engine to version `R1.00.0388.0243 (Oct 27 2024)`:
     - finally provide a 64bit native binary for all 64bit supported platforms (aarch64, x86_64) (c.f. #903).
     - enabled `_TIME_BITS=64` and removed hard limit <= year 2037 in the internal date check so that `time_t` values can carry 64bit signed values, thus dealing with dates with year >= 2038. This fixes certain Y2038 limitations for 64bit and 32bit versions of ReGaHss, thus make it Y2038+ aware.
     - changed script engine to use a 64bit integer datatype per default, thus e.g. integer values >2^31 can be used now.
     - add fix to correctly convert to an ISE object ID for certain script functions. Now several script functions can either take the object ID or directly derive it from the supplied object itself.
     - removed individual thread stack size settings and rely on the default stack sizes of the underlying operating system instead. This fixes certain crash issues with too low stack sizes in case of deep nested loop recursions.
     - fix issue with `.GetAt()` indexing not working correctly with 64bit integers.
   - With some dimmer links, a value entered as a percentage was displayed too small by a factor of 100 in expert mode.
   - HmIP-SWO (weather sensor) - The description for calculating the brightness threshold of the corresponding configuration parameter has been adjusted. Previously, the brightness threshold was set too low by the customer by a factor of 10.
   - HmIP-RGBW - When linking to a button, the value “S” (Saturation) was not applied to the color selection picker.
   - When using a system variable in %, this was output too small by a factor of 100.
   - HmIP-STV (tilt/vibration sensor) - The possible values for the configuration parameters angle and hysteresis for position detection and tilt detection have been adjusted.
   - The Easymode for devices of channel type COND_SWITCH_TRANSMITTER_TEMPERATURE, e.g. HmIP-STE-2-PCB, has been extended by 2 additional subsets:
     - Upper temperature threshold exceeded on / lower threshold undershot off
     - Upper temperature threshold exceeded off / lower threshold undershot on
   - The “Hours” option has been removed from the “Event delay” channel parameter
   - Easymodes for dimming actuators have been expanded to include the 'Minimum level in “on” state' parameter. This allows LED lamps to be switched on at lower values.
   - Thermostats now have the option of choosing between heating and cooling under Status/Operation.
   - Easymode added for linking the presence detector HmIP-SPI with HmIP-RGBW, HmIP-LSS and HmIP-DRG-DALI.
   - You can now filter by device under Status and operation.   
- [b676b5736](http://github.com/jens-maus/RaspberryMatic/commit/b676b5736ef0e41856f16f27da1817d7019e61f2) - fix `SYSTEM_HAS_USER_KEY` usage/setup in `restore_backup.cgi` script of recovery system.
- [5d3345f53](http://github.com/jens-maus/RaspberryMatic/commit/5d3345f53e3370ce438bfa6996653a7b6b05f620) - complete rework of `restoreBackup.sh` to not apply a supplied sbk backup right away but schedule it for the next boot cycle instead like a normal webui based backup restore is doing this. In addition, the new version now allows to perform certain consistency checks of a sbk backup and outputs this information on the command-line in addition to allowing to check if a supplied security key matches the backup and/or already installed system key.
- [b9479dd00](http://github.com/jens-maus/RaspberryMatic/commit/b9479dd00d408b57ad68b20ae8707d0ba68e0655) - enable firewall error logging per default and output as `user.err` instead.

#### WebUI changes:
- [d5df55afb](http://github.com/jens-maus/RaspberryMatic/commit/d5df55afb7fb1ec2e02bba829225796e862f9e6a) - Fix umlauts in `translate.lang.js` of `0136-WebUI-Add-Tailscale-VPN` WebUI patch (#2883, #2884, @boppbo)
- [b86c8b2bf](http://github.com/jens-maus/RaspberryMatic/commit/b86c8b2bf2fb77f51ebe6574872222b4e3913acd) - add another style glitch patch which brings back the more dense device list presentation in the device status display in OCCU 3.79.x.
- [b100e2c44](http://github.com/jens-maus/RaspberryMatic/commit/b100e2c44967d4ba072b0bd19f6f7c1516256e32) - remove of the `0024-WebUI-Filter-Operations` WebUI patch which has been integrated by eQ3 in OCCU 3.79.4. 
- [2a18bc977](http://github.com/jens-maus/RaspberryMatic/commit/2a18bc9779802dff803db48554242897d3ffd75f) - update CodeMirror script editor to latest 5.65.18 version.

#### Operating system changes:
- [beaed5276](http://github.com/jens-maus/RaspberryMatic/commit/beaed52767cf90f110cc5e55bf006dd1b573817e) - update rpi-eeprom to latest `pieeprom-2024-10-21.bin`
- [a7713a477](http://github.com/jens-maus/RaspberryMatic/commit/a7713a477990ff067e5bd92706aa707137c3a965) - add `noexec,nosuid,nodev` mount options to certain `tmpfs` mounts to ensure they are more safe to being misused. Also changed `inittab` to not directly mount but respect `fstab` settings/options.
- [655b9f697](http://github.com/jens-maus/RaspberryMatic/commit/655b9f6972160ebd63d6b45d1297445cf51a552e) - update upstream kernel to 6.6.58
- [2520ae049](http://github.com/jens-maus/RaspberryMatic/commit/2520ae0498181683aee116eea466e3bd01849a79) - bump qemu-guest-agent to 9.1.1
- [056307e76](http://github.com/jens-maus/RaspberryMatic/commit/056307e769388d489739ea198511aa3b366ea549) - update tailscale to latest v1.76.1 (#2882, @fl0x53).
- [fa328755b](http://github.com/jens-maus/RaspberryMatic/commit/fa328755bb6150fee3f2dbd74c54414528174f8a) - ensure that `/etc/init.d/S50crond` is removed in post-build.sh because we do have our own crond startup script in S98 (#2858).
- [30d9efe11](http://github.com/jens-maus/RaspberryMatic/commit/30d9efe11b55eae08e52c5eefdd8c8a224f587bd) - update buildroot to latest 2024.08.1
- [4f71d8b7b](http://github.com/jens-maus/RaspberryMatic/commit/4f71d8b7bfb1f020eb74e89ec5c801228305517f) - updated java azul to latest 11.76.21-ca-jre11.0.25
- [3ef3eb07e](http://github.com/jens-maus/RaspberryMatic/commit/3ef3eb07ef1dbca7291ea450a78883ee850b9f1e) - set `init_on_free=0` as default cmdline kernel option for generic-x86_64 to get certain newer NUC systems to boot correctly again when doing a UEFI boot (#2834).
- [d3d203baf](http://github.com/jens-maus/RaspberryMatic/commit/d3d203baf9ee81899cf9b19be12bf636c92dd0b6) - update rpi linux kernel to latest version (stable_20241008 / 6.6.51)
- [720613ae6](http://github.com/jens-maus/RaspberryMatic/commit/720613ae659dd6e467037cdccba1868d7b118fb3) - update rpi-firmware to latest version (1.20241008)
- [1a6b257ad](http://github.com/jens-maus/RaspberryMatic/commit/1a6b257ad6ed19158de6bc721930bd375bc6b77d) - add missing LIMIT and HF netfilter kernel extensions for improved ipv6 firewall support (#2874).

## 👪 Contributors (alphabetically):
- @boppbo
- @fl0x53
- @jens-maus

## 📝 Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## 📦 Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi5 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-rpi5.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-rpi5.zip)<br/>SHA256: 5f091b3fd2a57538a1e0c42a0dc665a9cb67f27a908b0994c30e6f243e9d2034</sub>

- RaspberryPi4 Model B, RaspberryPi Compute Module 4, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-rpi4.zip)<br/>SHA256: 66648525e0663c7355c6e6317967f7010e4cf64c96bbac3865d18e5463679ead</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite, RaspberryPi Zero 2 W – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-rpi3.zip)<br/>SHA256: 0d91ec1b6a58f9926e877643bc90489de02d24625aac7c7ebda9020f57057e03</sub>
<sub>📦 [RaspberryMatic-3.79.6.20241031-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: 41ff208d302f96739f083d485ab15051e8b0c3c9e07c3f098d87dd34d8a4e73a</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-rpi2.zip)<br/>SHA256: 878d8767b671821839f1ceffc02acc011083db1fd41df7c83d67131620f91756</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-rpi0.zip)<br/>SHA256: 82a8100501c79e6069986d301bcdd3521e011fd67b01e2560533ac07a75f7f66</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-tinkerboard.zip)<br/>SHA256: 4ac273567547d181ed46b9d38ddb6a3ad3cfd9b619c5f53185686adc13828ac8</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-odroid-n2.zip)<br/>SHA256: dceae989fff20b386ce634ccf4594acaef16f84ea11f7f916a23c8037dcab04f</sub>
<sub>📦 [RaspberryMatic-3.79.6.20241031-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-odroid-c4.zip)<br/>SHA256: 35d16b5ab5ba39239e302502ac97dd35449738d7d08d819fce74004e868ac946</sub>
<sub>📦 [RaspberryMatic-3.79.6.20241031-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-odroid-c2.zip)<br/>SHA256: 877e8383104d95b91dbbb767f1b97bd8ac8e86bb3b2c2db1386cdc436c99667d</sub>

- Generic-x86_64 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-generic-x86_64.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-generic-x86_64.zip)<br/>SHA256: 0eabcfef4be512fa2c1d0412365864af4005f96d2b74f10e32b5b552e8644106</sub>

- Open Virtual Appliance (OVA) – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [UNRAID](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-UNRAID), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-ova.zip)<br/>SHA256: dc5e4d0fa41957f176a7a3122740b37531195d734da967c0e2d7b19d46a58e9a</sub>
<sub>📦 [RaspberryMatic-3.79.6.20241031.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031.ova) (only for initial OVA installation)<br/>SHA256: 7c04787fdfcd2dc40ea69dbf4609ccc262531571e8701a6c0473b2d64df204aa</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: 40ed73a8dfc42d6f88c15d4c7927830a72c534b24fec2f9cd548857922e0af70</sub>
<sub>📦 [RaspberryMatic-3.79.6.20241031-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: c412f69e6910ad2c3539906eb049ec7c066b21633ad49cf2d9dfc8420c8be752</sub>
<sub>📦 [RaspberryMatic-3.79.6.20241031-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-oci_arm.tgz) (arm/armhf)<br/>SHA256: 78686502fee1eb2c36d7d6a5b22aa1ab7372d4c961206ce9e6c68b59f32667ed</sub>

- LXC Container – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-LXC)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-lxc_amd64.tar.xz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-lxc_amd64.tar.xz) (amd64/x86_64)<br/>SHA256: d3f589a4a28195086f612c14b77a1176a306949dd81f7bdca5bc686a94db9c84</sub>
<sub>📦 [RaspberryMatic-3.79.6.20241031-lxc_arm64.tar.xz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-lxc_arm64.tar.xz) (arm64/aarch64)<br/>SHA256: 814e5973411ef0961f679be9ce0b378c7f66878f99b85f1c7a4953ddcb256533</sub>
<sub>📦 [RaspberryMatic-3.79.6.20241031-lxc_arm.tar.xz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-lxc_arm.tar.xz) (arm/armhf)<br/>SHA256: a2ae1f4b752ba0c17999bf80848153436798ccd616922981aaa51871705b7db6</sub>

- Generic-aarch64 Appliance – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE)):
<sub>📦 [RaspberryMatic-3.79.6.20241031-generic-aarch64.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.79.6.20241031/RaspberryMatic-3.79.6.20241031-generic-aarch64.zip)<br/>SHA256: 4f65ed01797b1b79f502c047abc96a890f417ba32251e40b3104be036607cbd0</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.79.6.20241031'>RaspberryMatic 3.79.6.20241031</a>.</em>

**Comment by N3rdix** [comment]
Update went smooth as always 👍 

I was excited to see [HMCCU-1231] to switch between heat/cool for my thermostats (HmIP-BWTH) but it's not visible at all.
Channel 5 is already set to heat/cool...

Anyone had luck?

**Reply by fl0x53** [reply]
It's a bit odd for me, I have the buttons on 5 out of 10 thermostats. I couldn't figure out what the difference is...

**Reply by N3rdix** [reply]
Also odd:
I see the 2 buttons on the group of thermostats, but not on any single device...

**Comment by CB1277** [comment]
Update done. Works quiet well.

But I recogniced a problem, in programme and connections > programme and base connections you can select in the line action aktive, viewable and operable. When the last one will be deselected, it can´t be selected again.

Got anyone else that problem?

**Reply by jens-maus** [reply]
Show a screenshot and open a separate discussion thread if the issue is permanent.

**Reply by fl0x53** [reply]
Works for me 🤷

**Reply by Baxxy13** [reply]
Deactivate "operable" and activate it.
Reload program-list or log out of the WebUI, log in and check.

**Comment by tomturo245** [comment]
Update went well :-) as always. Only problem I found: the push messages from Pocket Control are not working anymore. Thought about downgrading to find out whether it is related to the new version. Then I saw the warning ** Breaking change ***

Does anyone else has problems with this? I changed my Iphone and hope it is not the root cause for the problem....
